pkg/api/norman/customization/cluster/validator.go

```diff
@@ -408,6 +413,61 @@ func (v *Validator) validateEKSConfig(request *types.APIContext, cluster map[str
 	return nil
 }
 
+// validateEKSKubernetesVersion checks whether a kubernetes version is provided and if it is supported
+func validateEKSKubernetesVersion(eksConfig map[string]interface{}, prevCluster *v3.Cluster, create bool) error {
+	clusterVersion, ok := eksConfig["kubernetesVersion"].(string)
+	if !ok {
+		return nil
+	}
+
+	for _, val := range availableVersions {
+		if clusterVersion == val {
+			return nil
+		}
+	}
+	return httperror.NewAPIError(httperror.InvalidBodyContent, fmt.Sprintf("invalid kubernetes version [%s]. Supported versions: %v", clusterVersion, availableVersions))
+}
+
+// validateEKSNodegroups checks whether a given nodegroup version is empty or not supported.
+// More involved validation is performed in the EKS-operator.
+func validateEKSNodegroups(eksConfig map[string]interface{}, prevCluster *v3.Cluster, create bool) error {
+	nodegroups, _ := eksConfig["nodeGroups"].([]interface{})
+	if len(nodegroups) == 0 {
+		return httperror.NewAPIError(httperror.InvalidBodyContent, fmt.Sprintf("must have at least one node group"))
+	}
+
+	var errors []string
+nodegroups:
+	for _, ng := range nodegroups {
+		ngMap, _ := ng.(map[string]interface{})
+		name, _ := ngMap["nodegroupName"].(string)
+		if name == "" {
+			return httperror.NewAPIError(httperror.InvalidBodyContent, fmt.Sprintf("nodegroupName cannot be an empty"))
+		}
+
+		ngVersion, ok := ngMap["version"].(string)
+		if !ok {
+			if create {
+				errors = append(errors, fmt.Sprintf("verison for nodegroup [%s] is required", name))
+			}
+			continue
+		}
+
+		for _, version := range availableVersions {
+			if ngVersion == version {
+				continue nodegroups
+			}
+		}
+		errors = append(errors, fmt.Sprintf("invalid kubernetes version [%s] for nodegroup [%s]. Supported versions: %v", ngVersion, name, availableVersions))
```

Comment:
Instead of appending error, shouldn't we fail-fast and error out as we find some invalid nodegroup?

Reply:
I feel it is better UX to receive all nodegroup version errors at once instead of resubmitting the request multiple times until all errors are resolved.